snowflake/data_access.go

```diff
@@ -388,7 +388,8 @@ func (s *AccessSyncer) importPoliciesOfType(accessProviderHandler wrappers.Acces
 	}
 
 	for _, policy := range policyEntities {
-		if !strings.EqualFold(strings.Replace(policyType, " ", "_", -1), policy.Kind) {
+		if !strings.HasPrefix(strings.Replace(policy.Kind, "_", " ", -1), policyType) {
```

Comment:
This is the actual change, all the rest of the PR is just updating the tests (and fixing a typo).